src/inference/src/cpp/core.cpp

```diff
@@ -137,6 +141,8 @@ CompiledModel Core::compile_model(const std::string& model,
                                   const std::string& device_name,
                                   const AnyMap& config) {
     OV_CORE_CALL_STATEMENT({
+        const auto device_not_supported = _impl->get_device_unsupported_properties(device_name, config);
+        set_property(device_not_supported);
```

Comment:
It's prohibited to change Core state via compile_model  call.

I think we need to split options to core / non-core inside _impl and ensure that core options are applied by core.

Reply:
Removed, maybe to functions `ov::Core::compile_model` should have `const` attribute to not allow this kind of implementation? The `CoreImpl` are const members.

Manage to parse inside the core.
But it looks like write-only properties has to be also added to supported properties (can be as internal on start or add some special group), because user config may want configure them on model compile (avoid exceptions when local config created).